cumulusci/tasks/salesforce/users/photos.py

```diff
@@ -0,0 +1,161 @@
+import re
+import base64
+import pathlib
+import json
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import CumulusCIException
+from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+
+def join_errors(e: SalesforceMalformedRequest) -> str:
+    return "; ".join([error.get("message", "Unknown.") for error in e.content])
+
+
+class UploadProfilePhoto(BaseSalesforceApiTask):
+    task_docs = """
+Uploads a profile photo for a specified or default User.
+
+Examples
+--------
+
+Upload a profile photo for the default user.
+
+.. code-block:: yaml
+
+    tasks:
+        upload_profile_photo_default:
+            group: Internal storytelling data
+            class_path: cumulusci.tasks.salesforce.users.UploadProfilePhoto
+            description: Uploads a profile photo for the default user.
+            options:
+                photo: storytelling/photos/default.png
+
+Upload a profile photo for a user whose Alias equals ``grace`` or ``walker``, is active, and created today.
+
+.. code-block:: yaml
+
+    tasks:
+        upload_profile_photo_grace:
+            group: Internal storytelling data
+            class_path: cumulusci.tasks.salesforce.users.UploadProfilePhoto
+            description: Uploads a profile photo for Grace.
+            options:
+                photo: storytelling/photos/grace.png
+                where: (Alias = 'grace' OR Alias = 'walker') AND IsActive = true AND CreatedDate = TODAY
+    """
+
+    task_options = {
+        "photo": {"description": "Path to user's profile photo.", "required": True},
+        "where": {
+            "description": """WHERE clause used querying for which User to upload the profile photo for.
+
+* No need to prefix with ``WHERE``
+
+* The SOQL query must return one and only one User record.
+
+* If no "where" is supplied, uploads the photo for the org's default User.
+
+""",
+            "required": False,
+        },
+    }
+
+    def _raise_cumulusci_exception(self, e: SalesforceMalformedRequest) -> None:
+        raise CumulusCIException(join_errors(e))
+
+    def _get_user_id_by_query(self, where: str) -> str:
+        # Query for the User removing a "WHERE " prefix from where if exists.
+        query = "SELECT Id FROM User WHERE {} LIMIT 2".format(
+            re.sub(r"^WHERE ", "", where, flags=re.I)
+        )
+        self.logger.info(f"Querying User: {query}")
+
+        user_ids = []
+        try:
+            for record in self.sf.query_all(query)["records"]:
```

Comment:
We have a `LIMIT 2`.  Does it matter using `self.sf.query` vs `self.sf.query_all`?

Reply:
I don't believe it does; there won't be a nextRecordsUrl ever.